libs/source/Source.mli

```diff
@@ -0,0 +1,77 @@
+(* Cooper Pierce
+ *
+ * Copyright (c) Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(** Sources of content which may be consumed by Semgrep.
+
+   A source is the {e user-relevant origin} of content (e.g., source code).
```

Comment:
I would either rename this file Src.ml, or maybe even better Origin.ml.
Note that @mjambon has actually a related type to what you did called Src_file.ml in the spacegrep codebase (libs/spacegrep/....).

Reply:
Renamed in 3f206d0db; I have seen Src_file.ml but as I understand it the purpose is a bit different, since I understand that to exist for the purpose of being able to persist a copy of the contents. `Src_file.source` is potentially replaceable with `Origin.t` at some point in the future.